### PR TITLE
Add edition to patterns

### PR DESCRIPTION
**Pull request by dreulavelle**
This could possibly use some slight tweaking, but it works well. Let me know if anything needs changed! 

Added 2 tests as well.

Ex:

"The.Lord.of.the.Rings.The.Fellowship.of.the.Ring.2001.EXTENDED.2160p.UHD.BluRay.x265.10bit.HDR.TrueHD.7.1.Atmos-BOREDOR"

```json
{
  "resolution": "2160p",
  "quality": "Blu-ray",
  "year": 2001,
  "codec": "H.265",
  "audio": "Dolby TrueHD 7.1",
  "extended": true,
  "bitDepth": 10,
  "hdr": true,
  "edition": "Extended Edition",
  "title": "The Lord of the Rings The Fellowship of the Ring",
  "encoder": "BOREDOR"
}
```

### CODE REVIEW

**Comment by mhdzumair** [comment]
@dreulavelle update the PR target branch as dev

**Comment by dreulavelle** [comment]
Done, my apologies!

**Comment by dreulavelle** [comment]
Feel free to edit the regex as you like.

File: PTN/patterns.py [hunk 1]
```diff
@@ -350,6 +351,20 @@
     + patterns["subtitles"][-2]
     + ")",
 ]
+
+patterns["edition"] = [
+    (r"\b\d{{2,3}}(th)?{d}Anniversary{d}(Edition|Ed)?\b".format(d=delimiters), "Anniversary Edition"),
+    (r"\bUltimate{d}Edition\b".format(d=delimiters), "Ultimate Edition"),
+    (r"\bExtended{d}Director\"?s\b".format(d=delimiters), "Directors Cut"), # Needs to be before "Extended"
+    (r"\bExtended\b", "Extended Edition"),
+    (r"\bDirector\"?s{d}Cut\b".format(d=delimiters), "Directors Cut"),
```
Comment (mhdzumair) [inline PTN/patterns.py:367]: you may combine the "Directors Cut"

Comment (dreulavelle) [reply PTN/patterns.py:367]: I can fix this tonight, thanks! Did you just want it on one line separated with a pipe? Or just cover `director` instead of both edge cases?

Comment (dreulavelle) [reply PTN/patterns.py:367]: Ah I'll combine, checking `extended` first

Comment (mhdzumair) [reply PTN/patterns.py:367]: one line separated with a pipe

**Comment by platelminto** [comment]
Hey! Thanks for opening the PR.

Extended editions & directors cuts are already fields: `extended` and `directorsCut`. If the current patterns aren't matching something you'd like to match, you can add to those, and add the relevant titles as tests. I'm unsure if an extended director's cut would currently match for both fields, and if you can provide an example title for this, we can go from there.

As for "Ultimate" and "Anniversary" editions, we can add these if there are enough example titles to work from - I don't like adding new fields unless I see several examples of them from real torrents. If you can provide these (just the titles though! don't include torrent URLs here), we can think about adding them. This would likely be as 2 separate fields, similar to `extended` and `directorsCut`.

Also, please make sure all tests pass when opening a PR! If you had run them, you might have noticed the duplicate fields. Maybe re-read [Contributing](https://github.com/platelminto/parse-torrent-title#contributing). Additionally, the second test you provided doesn't make sense - the output given doesn't match the input. Based on the input though, I've improved the `extended` pattern, and added an adapted version of the title to the tests. `dev` should already contain these changes. Unfortunately, the full collection title will still struggle to process correctly due to #20.

Thanks again, and let me know your thoughts!
